Example/Swift/ViewController.swift

```diff
@@ -113,22 +139,85 @@ class ViewController: UIViewController, MGLMapViewDelegate, AVSpeechSynthesizerD
-            var routeCoordinates = route.coordinates!
-            let line = MGLPolyline(coordinates: &routeCoordinates, count: route.coordinateCount)
-            self?.mapView.addAnnotation(line)
+            guard let style = self?.mapView.style else {
+                return
+            }
+            
+            self?.userRoute = route
+            self?.toggleNavigationButton.isHidden = false
+            
+            // Remove old destination marker
+            self?.removeRoutesFromMap()
+            
+            // Add destination marker
+            let destinationMarker = MGLPointAnnotation()
+            destinationMarker.coordinate = route.coordinates!.last!
+            self?.mapView.addAnnotation(destinationMarker)
             
-            self?.startNavigation(route)
+            let polyline = MGLPolylineFeature(coordinates: route.coordinates!, count: route.coordinateCount)
+            let geoJSONSource = MGLShapeSource(identifier: sourceIdentifier, shape: polyline, options: nil)
+            let line = MGLLineStyleLayer(identifier: layerIdentifier, source: geoJSONSource)
+            
+            
+            // Style the line
+            line.lineColor = MGLStyleValue(rawValue: UIColor(red:0.00, green:0.45, blue:0.74, alpha:0.9))
+            line.lineWidth = MGLStyleValue(rawValue: 5)
+            line.lineCap = MGLStyleValue(rawValue: NSValue(mglLineCap: .round))
+            line.lineJoin = MGLStyleValue(rawValue: NSValue(mglLineJoin: .round))
+
+            // Add source and layer
+            style.addSource(geoJSONSource)
+            for layer in style.layers.reversed() {
+                if !(layer is MGLSymbolStyleLayer) {
+                    style.insertLayer(line, above: layer)
+                    return
+                }
+            }
         }
     }
     
     func startNavigation(_ route: Route) {
+        let camera = mapView.camera
+        camera.pitch = 40
+        mapView.setCamera(camera, animated: false)
         mapView.userTrackingMode = .followWithCourse
         navigation = RouteController(route: route)
         navigation?.resume()
     }
+    
+    func endNavigation() {
+        instructionView.isHidden = true
+        toggleNavigationButton.isHidden = true
+        mapView.userTrackingMode = .none
+        let camera = mapView.camera
+        camera.pitch = 0
+        mapView.setCamera(camera, animated: true)
+        removeRoutesFromMap()
+        navigation?.suspend()
+    }
+    
+    func removeRoutesFromMap() {
+        guard let style = mapView.style else {
+            return
+        }
+        if let line = style.layer(withIdentifier: layerIdentifier) {
+            style.removeLayer(line)
+        }
+        if let source = style.source(withIdentifier: sourceIdentifier) {
+            style.removeSource(source)
+        }
+        mapView.removeAnnotations(mapView.annotations ?? [])
+    }
+    
+    func roundToTens(_ x: CLLocationDistance) -> Int {
```

Comment:
If you’re OK with requiring iOS 10 for this example application, you can use a MeasurementFormatter, which exposes a `numberFormatter` property. The NumberFormatter class exposes a few different options for rounding. I think we should recommend the use of MeasurementFormatter via this example, since U.S.-based developers will naturally want to support whatever units the system uses.